pkg/cmd/server/start/start_master.go

```diff
@@ -513,20 +515,21 @@ func startControllers(oc *origin.MasterConfig, kc *kubernetes.MasterConfig) erro
 	oc.RunSecurityAllocationController()
 
 	if kc != nil {
+		// ensure pending namespace deletions are not blocked by later controller startup failures
+		kc.RunNamespaceController()
+		// called by admission control
+		kc.RunResourceQuotaManager()
+
 		_, rcClient, err := oc.GetServiceAccountClients(oc.ReplicationControllerServiceAccount)
```

Comment:
While this change now gives the `NamespaceController` a shot at running, you're still hoping that an asynchronous worker thread manages to get some work done before it gets here.  It's entirely possible that it can't even complete the list operation before it gets here.

Perhaps now is the time to build in the logic to relocate SA credentials on 401 failures from controllers.